docs/configuration/05-autocommands.md

```diff
@@ -1,17 +1,54 @@
 # Autocommands
 
-To set up autocommands use `lvim.autocommands.custom_groups`. Autocommands are defined in the form `{ Event, filetype, command }`. This will run a command at a given event for the given filetype.
-
-To view help on autocommands: `:h autocmd`
-
-Here is a [list of events](https://tech.saigonist.com/b/code/list-all-vim-script-events.html)
+To set up autocommands use the native nvim api `vim.api.nvim_create_autocmd` or the Lunarvim table `lvim.autocmds` to load a list of autocommands.
+Autocommands are defined in the form
+```lua
+lvim.autocmds = {
+  { -- first entry
+    <EVENT>,
+    { -- this is passed directly as opts to `nvim_create_autocmd()`
+      pattern = { <FILETYPE> },
+      -- enable wrap mode for json files only
+      command = "<COMMAND>",
+    },
+  },
+  -- add more entries
+}
+```
+an concrete example could look like this:
+```lua
+lvim.autocmds = {
+    "BufEnter",
+      pattern = { "*.json", "*.jsonc" },
+      command = "setlocal wrap",
+    },
+  },
+}
+```
+This will run a command at a given event for the given filetype.
 
+An example using the nvim api could look like this:
 ```lua
-lvim.autocommands.custom_groups = {
-  -- On entering a lua file, set the tab spacing and shift width to 8
-  { "BufWinEnter", "*.lua", "setlocal ts=8 sw=8" },
+vim.api.nvim_create_autocmd("BufEnter", {
+	  pattern = { "*.json", "*.jsonc" },
+	  -- enable wrap mode for json files only
+	  command = "setlocal wrap",
+})
+```
+You can also add custom mappings using `command`. For example the following autocommand would add the mapping `<space><space>` to `cpp` and `hpp` files to switch between source and header files using `ClangdSwitchSourceHeader`:
```

Comment:
I think the `ClangdSwitchHeader` example is better addressed within the LSP utilities, e.g. `on_attach` or the new `LspAttach` event. 

relevant: https://www.lunarvim.org/languages/c_cpp.html#lsp-settings

Reply:
At first I had the typescript example in here (from the liked issue) but then I thought it might be easier to have a single command in there, because I think it is easier to understand. The callback example looks good too but I think a mapping example using commands should be listed there as well. Why not do both, there is not that much content on the autocommand page right now.

Reply:
you can add it if you want, but just use lua to set the keybind instead, just to make it more familiar. I would've used `vim.opt_local`, but I don't think it's that reliable.